### PR TITLE
fix: Allow to add more than one VNetCIDR to the exception list

### LINKED ISSUE #3789: OutBoundNAT ExceptionList is only populated with the first entry from the list of Address Spaces of the VNET of the cluster
In situations where there are multiple address ranges defined for a VNET there's only the first CIDR taken into the OutBoundNAT exclusion list on Windows Nodes preventing the network communication with services that are in network ranges that are outside of the first CIDR exposed in pods running on the windows node pool.

*Workaround for two CIDR*
Have the CIDR that is not used as the node pool range as the first entry as the node pool range is always added as an exclusion

**Steps To Reproduce**
 - Create a VNET with at least two CIDRs
 - Provision the AKS into the first CIDR with a windows node pool
 - Deploy a pod hosting a website into the windows node pool
 - Provision an Application Gateway into the 2nd range
 - The Application Gateway is not able to reach services (i.e. a website) that is exposed from the windows pod

**Expected behavior**
- The Application Gateway is able to reach the service exposed by the windows pod.
- All CIDRs that are defined in the VNET that the AKS cluster is bound to are added to the ExceptionList of the OutBoundNAT on Windows nodes

**AKS Engine version**
v0.47.0-aks-gomod-a8-aks

**Kubernetes version**
1.18.6

**Additional context**
I believe that this assumption that is stated in the code is wrong and therefor it doesn't behave correct.

problemInCode

I can unfortunately not find that any longer (I detected the issue a while ago and still had the creation of the issue pending)

### PR DESCRIPTION
**Pull request by AbelHu**
fix #3789
**Reason for Change**:

**Issue Fixed**:

**Requirements**:

- [x] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [ ] includes documentation
- [ ] adds unit tests
- [ ] tested upgrade from previous version

**Notes**:

### CODE REVIEW

**Comment by jsturtevant** [comment]
Is this something that should go into the main branch?

**Comment by marosset** [comment]
Should we add this to master as well?

**Comment by AbelHu** [comment]
> Should we add this to master as well?

Yes. Will submit another PR later.

File: parts/k8s/windowsazurecnifunc.ps1 [hunk 1]
```diff
@@ -76,10 +76,28 @@ Set-AzureCNIConfig
     # Fill in DNS information for kubernetes.
     if ($IsDualStackEnabled){
         $subnetToPass = $KubeClusterCIDR -split ","
-        $exceptionAddresses = @($subnetToPass[0], $MasterSubnet, $VNetCIDR)
+        $exceptionAddresses = @($subnetToPass[0], $MasterSubnet)
+
+        if ($VNetCIDR.Contains(",")) {
```
Comment (mainred) [inline parts/k8s/windowsazurecnifunc.ps1:81]: Maybe we can do `-split ","` anyway.
```
PS C:\Users\qinhao> $VNetCIDR="192.168.0.0/24"
PS C:\Users\qinhao> $VNetCIDR -split ","
192.168.0.0/24
PS C:\Users\qinhao> $VNetCIDR="192.168.0.0/24,192.168.1.0/24"
PS C:\Users\qinhao> $VNetCIDR -split ","
192.168.0.0/24
192.168.1.0/24
```

File: parts/k8s/windowsazurecnifunc.ps1 [hunk 2]
```diff
@@ -76,10 +76,28 @@ Set-AzureCNIConfig
...
+            }
+        } else {
+            $exceptionAddresses += $VNetCIDR
+        }
     }
     else {
-        $exceptionAddresses = @($KubeClusterCIDR, $MasterSubnet, $VNetCIDR)
+        $exceptionAddresses = @($KubeClusterCIDR, $MasterSubnet)
+
+        if ($VNetCIDR.Contains(",")) {
```
Comment (mainred) [inline parts/k8s/windowsazurecnifunc.ps1:93]: ditto

**Review by mainred — APPROVED** [review]
LGTM, thank you.

**Pull request merged**